mlir/lib/Conversion/TosaToTensor/TosaToTensor.cpp

```diff
@@ -268,12 +268,28 @@ class SliceConverter : public OpConversionPattern<tosa::SliceOp> {
     ShapedType resultType = cast<ShapedType>(sliceOp.getType());
     if (llvm::isa<UnrankedTensorType>(resultType))
       return failure();
+
+    ElementsAttr StartElems;
+    ElementsAttr SizeElems;
+
+    if (!matchPattern(sliceOp.getStart(), m_Constant(&StartElems)))
```

Comment:
Non-static legalization? Do we expect this to come to a separate patch?

Reply:
cc original author @tatwaichong

Reply:
Sorry, I don't get this question? Can you expand on it?
`tosa-to-tensor-invalid.mlir` and `tosa-to-tensor.mlir` have lowering tosa.slice to tensor ops. I think we have to adjust the tosa-to-tensor conversion and its lit tests when the argument types of tosa.slice are changed, and do it in the same patch.

Reply:
So from what I gather legalizations for when the parameters are not constant will be done in a separate patch sometime in the future?

Reply:
I see. As you have gathered, the dynamic-shaped handling has been implemented internally but the needed shape ops are not included in v1.0 and not published. Yes, non-static legalization will come to a separate patch sometime in the future, but I think the ultimate implementation may different with the current experimental IR when we found better design.

Reply:
Ack